tests/testthat/test-bias.R

```diff
@@ -69,37 +67,126 @@ test_that("bias works", {
+test_that("bias_quantile() returns 1 if true value below min prediction", {
+  predictions <- c(2, 3, 4, 5)
+  quantiles <- c(0.1, 0.3, 0.7, 0.9)
+  
+  suppressMessages(
+    expect_equal(bias_quantile(predictions, quantiles, true_value = 1), 1)
+  )
+})
 
-  scoringutils2 <- bias_range(
-    lower = lower, upper = upper,
-    range = range, true_value = true_value
+test_that("bias_quantile() returns -1 if true value above max prediction", {
+  predictions <- c(1, 2, 3, 4)
+  quantiles <- c(0.1, 0.3, 0.5, 0.7)
+
+  expect_equal(bias_quantile(predictions, quantiles, true_value = 6), -1)  
+})
+
+test_that("bias_quantile(): quantiles must be between 0 and 1", {
+  predictions <- 1:4
+  
+  # Failing example
+  quantiles <- c(-0.1, 0.3, 0.5, 0.8)
+  expect_error(bias_quantile(predictions, quantiles, true_value = 3),
+               "quantiles must be between 0 and 1")
+               
+  # Passing counter example               
+  quantiles <- c(0.1, 0.3, 0.5, 0.8)
+  expect_silent(bias_quantile(predictions, quantiles, true_value = 3))
+})
+
+test_that("bias_quantile(): quantiles must be increasing", {
+  predictions <- 1:4
+  
+  # Failing example
+  quantiles <- c(0.8, 0.3, 0.5, 0.9)
+  expect_error(bias_quantile(predictions, quantiles, true_value = 3),
+               "quantiles must be increasing")
+               
+  # Passing counter example
+  quantiles <- c(0.3, 0.5, 0.8, 0.9)
+  expect_silent(bias_quantile(predictions, quantiles, true_value = 3))
+})
+
+test_that("bias_quantile(): quantiles must be unique", {
+  predictions <- 1:4
+  
+  # Failing example
+  quantiles <- c(0.3, 0.3, 0.5, 0.8) 
+  expect_error(bias_quantile(predictions, quantiles, true_value = 3),
+               "quantiles must be increasing")
+               
+  # Passing example
+  quantiles <- c(0.3, 0.5, 0.8, 0.9)
+  expect_silent(bias_quantile(predictions, quantiles, true_value = 3))
+})
+
+test_that("bias_sample() approx equals bias_quantile() for many samples", {
+  set.seed(123)
+
+  # Generate true value
+  true_value <- 3
+  
+  # Generate many sample predictions
+  predictions <- sample(rnorm(1000, mean = true_value, sd = 2), 1000)
+
+  # Get sample based bias
+  bias_sample_result <- bias_sample(
+    true_value, matrix(predictions, nrow = 1)
   )
-  scoringutils <- bias_range(
-    lower = lower, upper = upper,
-    range = range, true_value = true_value
+
+  # Convert predictions to quantiles
+  quantiles <- seq(0, 1, length.out = 100)
+  quantile_preds <- quantile(predictions, probs = quantiles)
+
+  # Get quantile based bias
+  bias_quantile_result <-   suppressMessages(
+    bias_quantile(quantile_preds, quantiles, true_value)
   )
 
-  expect_equal(scoringutils, scoringutils2)
+  # Difference should be small
+  expect_equal(bias_quantile_result, bias_sample_result, tolerance = 0.1)
 })
 
-test_that("quantile bias and range bias have the same result", {
+test_that("bias_quantile and bias_range() give the same result", {
   predictions <- order(rnorm(23))
```

Comment:
Should this be `sort()` instead of `order()`? 

I just tried this and got

> predictions <- order(rnorm(23))
> lower <- rev(predictions[1:12])
> upper <- predictions[12:23]
> lower
 [1] 21 13  9  8  6 18 12  7 17 22 14  4
> upper
 [1] 21 15  1 11 16  5  2  3 10 20 23 19

Reply:
Used your suggestion in https://github.com/epiforecasts/scoringutils/pull/305/commits/06d1fe0ffe7dce8dcacc03a5242af90cfcf2f114 which addresses this.